presto-parquet/src/main/java/io/prestosql/parquet/writer/repdef/DefLevelIterable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.parquet.writer.repdef;
+
+import com.google.common.collect.AbstractIterator;
+
+import java.util.Optional;
+
+public interface DefLevelIterable
+{
+    DefLevelIterator getIterator();
+
+    abstract class DefLevelIterator
+            extends AbstractIterator<Optional<Integer>>
```

Comment:
Maybe use `OptionalInt` here to avoid double boxing.... only do this if it is convenient for he callers